src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -349,19 +420,33 @@ export class HttpConnection implements IConnection {
         return transport && typeof (transport) === "object" && "connect" in transport;
     }
 
-    private changeState(from: ConnectionState, to: ConnectionState): boolean {
-        if (this.connectionState === from) {
-            this.connectionState = to;
-            return true;
-        }
-        return false;
-    }
-
     private stopConnection(error?: Error): void {
+        this.logger.log(LogLevel.Debug, `HttpConnection.stopConnection(${error}) called while in sate ${this.connectionState}.`);
+
+        const connectionStarted = this.connectionStarted;
+
         this.transport = undefined;
+        this.connectionStarted = false;
 
         // If we have a stopError, it takes precedence over the error from the transport
         error = this.stopError || error;
+        this.stopError = undefined;
+
+        if (this.connectionState === ConnectionState.Disconnected) {
+            this.logger.log(LogLevel.Debug, `Call to HttpConnection.stopConnection(${error}) was ignored because the connection is already in the disconnected state.`);
+            return;
+        }
+
+        if (this.connectionState === ConnectionState.Connecting) {
```

Comment:
Is this the correct behavior? If `stopConnection `gets called while we're in the connecting state should that override connecting?

Reply:
If `stopConnection` was called, we'd be in the Disconnecting state.

Reply:
This is what the post-startInternal logic in start() is taking care of. By the time start exits, the state will be Disconnected.